libgalois/src/analytics/subgraph_extraction/subgraph_extraction.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file belongs to the Galois project, a C++ library for exploiting
+ * parallelism. The code is being released under the terms of the 3-Clause BSD
+ * License (a copy is located in LICENSE.txt at the top-level directory).
+ *
+ * Copyright (C) 2019, The University of Texas at Austin. All rights reserved.
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include "katana/analytics/subgraph_extraction/subgraph_extraction.h"
+
+#include "katana/TypedPropertyGraph.h"
+#include "katana/analytics/Utils.h"
+
+using namespace katana::analytics;
+
+namespace {
+using edge_iterator = boost::counting_iterator<uint64_t>;
+
+katana::Result<std::unique_ptr<katana::PropertyGraph>>
+SubGraphNodeSet(
+    katana::PropertyGraph* graph, const std::vector<uint32_t>& node_set) {
+  auto subgraph = std::make_unique<katana::PropertyGraph>();
+  if (node_set.empty()) {
+    return std::unique_ptr<katana::PropertyGraph>(std::move(subgraph));
+  }
+
+  uint64_t num_nodes = node_set.size();
+  // Subgraph topology : out indices
+  auto out_indices = std::make_unique<katana::LargeArray<uint64_t>>();
+  out_indices->allocateInterleaved(num_nodes);
+
+  katana::gstl::Vector<katana::gstl::Vector<uint32_t>> subgraph_edges;
+  subgraph_edges.resize(num_nodes);
+
+  katana::GAccumulator<uint64_t> accum_num_edges;
+  katana::do_all(
+      katana::iterate(node_set),
+      [&](const uint32_t& n) {
+        for (auto dest : node_set) {
```

Comment:
This is fine for now. But an optimization could be to sort this set and then intersection this list with the neighbors list.